theano/tensor/extra_ops.py

```diff
@@ -647,9 +647,36 @@ def repeat(x, repeats, axis=None):
 
     .. versionadded:: 0.6
     """
-    return RepeatOp(axis=axis)(x, repeats)
+    if isinstance(repeats, (int, long, numpy.integer, list)):
+        repeats = numpy.asarray(repeats)
```

Comment:
`repeats` could be a `tensor.vector()`, in that case, you cannot cast it as a numpy array.
It could also be a symbolic scalar.